app/src/main/java/com/realityexpander/tasky/agenda_feature/presentation/agenda_screen/AgendaScreen.kt

```diff
@@ -510,17 +554,61 @@ fun AgendaScreenContent(
             }
         }
 
-        // • FAB to add new agenda item
+
+        // • Create AgendaItem Event/Task/Reminder
+        DropdownMenu(
+            expanded = state.isAddAgendaItemDropdownVisible,
+            onDismissRequest = { onAction(AgendaEvent.ToggleCreateAgendaItemDropdown) },
+            offset = DpOffset(
+                x = fabButtonOffset.x.dp,
+                y = -screenHeight + fabButtonOffset.y.dp
+            ),
+            properties = PopupProperties(
+                usePlatformDefaultWidth = true,
+            ),
+            modifier = Modifier
+                .background(color = MaterialTheme.colors.onSurface)
+        ) {
```

Comment:
Little tip: You can put your `DropdownMenu` in a box with the view it should be anchored to (like the FAB here) and it'll automatically open above/below it. For example, this is my dropdown for logging out:
```
Box {
    DropdownMenu(
        expanded = state.isSelectingUserOptions,
        onDismissRequest = {
            viewModel.onToggleUserOptions(false)
        }
    ) {
        DropdownMenuItem(onClick = viewModel::logout) {
            Text(text = stringResource(id = R.string.logout))
        }
    }
    UserAvatar(
        avatarInitials = state.userInitials,
        modifier = Modifier
            .size(40.dp),
        onAvatarClick = {
            viewModel.onToggleUserOptions(true)
        }
    )
}
```

Reply:
OK, i will try out this implementation. It should save a lot of confusing code!